back/app/models/concerns/post.rb

```diff
@@ -55,8 +55,8 @@ module Post
 
     scope :order_new, ->(direction = :desc) { order(published_at: direction) }
     scope :order_random, lambda { |user|
-      modulus = RandomOrderingService.new.modulus_of_the_day(user)
-      order(Arel.sql("(extract(epoch from #{table_name}.created_at) * 100)::bigint % #{modulus}, #{table_name}.id"))
+      hash_part_for_today_and_user = Time.zone.today.to_s + user&.id.to_s
+      order(Arel.sql("md5(concat(id, '#{hash_part_for_today_and_user}'))"))
```

Comment:
Given the number of items we're ever likely to be returning and the size of data we're hashing I don't think performance of md5 will be of concern, but just flagging anyway.